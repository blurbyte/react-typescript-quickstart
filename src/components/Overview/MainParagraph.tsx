import styled from '@styles/styledComponents';
import { Colors } from '@styles/colors';

const MainParagraph = styled.p`
  font-size: 2.4rem;
  color: ${Colors.darkGray};
  line-height: 1.4;
  margin: 0.8rem 0;
`;

export default MainParagraph;
