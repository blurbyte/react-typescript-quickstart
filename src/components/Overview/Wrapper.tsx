import styled from '@styles/styledComponents';

const Wrapper = styled.section`
  grid-column: 1 / -1;
`;

export default Wrapper;
