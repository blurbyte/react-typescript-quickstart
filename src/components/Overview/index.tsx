import * as React from 'react';

import MainParagraph from '../MainParagraph';
import Headline from '../Headline';
import Wrapper from './Wrapper';

class Overview extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <Wrapper>
        <Headline>Welcome to React TypeScript Quickstart</Headline>
        <MainParagraph>
          A simple React starter kit, but with TypeScript. It takes care of initial developement environment setup and
          puts everything to use in simple example. Notable technologies used: React, TypeScript, FuseBox, TSLint and
          styled-components. Formatted with prettier.
        </MainParagraph>
      </Wrapper>
    );
  }
}

export default Overview;
