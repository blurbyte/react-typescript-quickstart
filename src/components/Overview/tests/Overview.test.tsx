import * as React from 'react';
import { shallow } from 'enzyme';

import Overview from '../index';

describe('Overview', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Overview />);
    expect(wrapper).toMatchSnapshot();
  });
});
