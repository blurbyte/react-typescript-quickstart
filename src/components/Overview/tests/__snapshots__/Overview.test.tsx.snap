// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview renders correctly 1`] = `
<styled.section>
  <styled.h1>
    Welcome to React TypeScript Quickstart
  </styled.h1>
  <styled.p>
    A simple React starter kit, but with TypeScript. It takes care of initial developement environment setup and puts everything to use in simple example. Notable technologies used: React, TypeScript, FuseBox, TSLint and styled-components. Formatted with prettier.
  </styled.p>
</styled.section>
`;
